features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/notifications/NotificationSettingsPresenter.kt

```diff
@@ -68,6 +76,61 @@ class NotificationSettingsPresenter @Inject constructor(
             observeNotificationSettings(matrixSettings)
         }
 
+        // List of PushProvider -> Distributor
+        val distributors = remember {
+            pushService.getAvailablePushProviders()
+                .flatMap { pushProvider ->
+                    pushProvider.getDistributors().map { distributor ->
+                        pushProvider to distributor
+                    }
+                }
+        }
+        // List of Distributor names
+        val distributorNames = remember {
+            distributors.map { it.second.name }
```

Comment:
Maybe use `toImmutableList()` here instead of in the return so it's only done once?

Reply:
Sorry I do not understand your remark.

Reply:
In [line 168](https://github.com/element-hq/element-x-android/pull/2928/files#diff-992aff3ec5cbd1d0a22fae581a1bb1929dfc078d1ffc22711782f4878c4e8822R168) we get this and apply `.toImmutableList()` in every recomposition. Since it won't change ever, it's better to just do it inside the remember, right?